internal/core/domain/deposit_repository.go

```diff
@@ -4,10 +4,8 @@ import "context"
 
 type DepositRepository interface {
 	AddDeposit(ctx context.Context, deposit Deposit) error
-	ListDepositsForAccountIdAndPage(
-		ctx context.Context,
-		accountIndex int,
-		page Page,
+	ListDepositsForAccountId(
+		ctx context.Context, accountIndex int, page *Page,
 	) ([]Deposit, error)
-	ListAllDeposits(ctx context.Context) ([]Deposit, error)
+	ListAllDeposits(ctx context.Context, page *Page) ([]Deposit, error)
```

Comment:
Based on name all objects should be retrieved and therefor Page object shouldnot be passed as arg.